lib/evmone_precompiles/ripemd160.cpp

```diff
@@ -0,0 +1,219 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2024 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ripemd160.hpp"
+#include <algorithm>
+#include <array>
+#include <bit>
+#include <cstdint>
+#include <utility>
+
+#if defined(_LIBCPP_VERSION) && _LIBCPP_VERSION < 180000
+// libc++ before version 18 has incorrect std::rotl signature
+// https://github.com/llvm/llvm-project/commit/45500fa08acdf3849de9de470cdee5f4c8ee2f32
+#pragma clang diagnostic ignored "-Wsign-conversion"
+#endif
+
+namespace evmone::crypto
+{
+namespace
+{
+// TODO(C++23): Use std::byteswap.
```

Comment:
consider this:
https://github.com/erigontech/silkworm/blob/master/silkworm/db/snapshots/seg/compressor/huffman_code.cpp#L176

I've used `__cplusplus` instead of `TODO(C++23)` and it is possible to use intrinsics if you know `sizeof(T)`.

Reply:
I prefer to not keep TODOs in code if possible due to https://cwe.mitre.org/data/definitions/546

If you want to avoid people trying C++23, it is better to disable it in the global makelist, and in the error message explain how to disable it if they want to still use C++23 at their own risk (maybe with an extra flag).

Using `__cplusplus` feels simpler to my taste, because once we update our baseline, which is likely to happen, it will be easier to clean up the code.

Reply:
The `TODO(C++23)` comments mean: after C++23 is enabled you can come here and simplify/improve the code. I actually did this for C++20 and this is not much work but can discover still missing features in compilers.

Maybe we can use different tag than TODO, but it won't solve the CWE issue.